src/Cinnabari.php

```diff
@@ -39,10 +39,16 @@ public function __construct(Schema $schema)
     // TODO: trim the query string before Cinnabari
     public function translate($query, $arguments)
     {
-        $tokens = self::getTokens($query);
-        $request = self::getRequest($tokens);
+        try {
+            $tokens = self::getTokens($query);
+            $request = self::getRequest($tokens);
+
+            return self::getResult($this->schema, $request, $arguments);
+        } catch (Exception $exception) {
+            echo $exception->getMessage() . "\n";
```

Comment:
We can keep the try-catch block here or move it to the private repository: Either way is fine, as long as we catch all of our exceptions at some point within our codebase. (Otherwise, we would cause a fatal error in the portal code.)

We should remove all echo statements, even if it was me who added it. Echo statements will be an issue in production, because they will corrupt the JSON endpoint and cause to appear on the partner-visible portal pages.